packages/nimble-components/src/table/models/table-layout-manager.ts

```diff
@@ -0,0 +1,353 @@
+
+    public handleChange(source: unknown, args: unknown): void {
+        if (
+            source instanceof Virtualizer
+            && args === 'headerContainerMarginRight'
+        ) {
+            this.updateTableViewportMinWidth();
+        }
+    }
+
+    private readonly onDividerMouseMove = (event: Event): void => {
+        const mouseEvent = event as MouseEvent;
+        const deltaX = mouseEvent.movementX > 0
+            ? Math.floor(mouseEvent.movementX)
+            : Math.ceil(mouseEvent.movementX);
+        this.currentTotalDelta += deltaX;
+        for (let i = 0; i < this.table.columns.length; i++) {
+            this.table.columns[i]!.columnInternals.currentPixelWidth = this.initialColumnPixelWidths[i]?.initialPixelWidth;
+        }
+        this.currentTotalDelta = this.pinColumnSizeDelta(
+            this.activeColumnDivider!,
+            this.currentTotalDelta
+        );
+        this.performCascadeSizeLeft(
+            this.activeColumnDivider!,
+            this.currentTotalDelta
+        );
+        this.performCascadeSizeRight(
+            this.activeColumnDivider!,
+            this.currentTotalDelta
+        );
+
+        this.table.tableWidthFactor = Math.max(
+            this.getCurrentTotalTableWidth() / this.initialTableWidth!,
+            1
+        );
+    };
+
+    private readonly onDividerMouseUp = (): void => {
+        document.removeEventListener('mousemove', this.onDividerMouseMove);
+        document.removeEventListener('mouseup', this.onDividerMouseUp);
+        this.unflagActiveColumnDividers();
+        this.updateTableViewportMinWidth();
+        this.resetGridSizedColumns();
+        this.table.isColumnBeingSized = false;
+    };
+
+    private getAllColumnsMinimumWidth(): number {
+        let totalColumnMiniumWidth = 0;
+        for (const column of this.table.columns) {
+            totalColumnMiniumWidth += column.columnInternals.minPixelWidth;
+        }
+        return totalColumnMiniumWidth;
+    }
+
+    private getTotalColumnFixedWidth(): number {
+        let totalColumnFixedWidth = 0;
+        for (const column of this.table.columns) {
+            totalColumnFixedWidth += (column.columnInternals.currentPixelWidth
+                !== undefined)
+                ? column.columnInternals.currentPixelWidth
+                : 0;
+        }
+        return totalColumnFixedWidth;
+    }
+
+    private getCurrentTotalTableWidth(): number {
+        return (
+            this.initialTableScrollableWidth!
+            + (this.getTotalColumnFixedWidth() - this.initialColumnTotalWidth!)
+        );
+    }
+
+    private setColumnsToFixedSize(): void {
+        this.cacheGridSizedColumns();
+        const headers = this.table.columnHeadersContainer.querySelectorAll(
+            '.header-container'
+        );
+        for (let i = 0; i < headers.length; i++) {
+            this.table.columns[i]!.columnInternals.currentPixelWidth = headers[i]!.getBoundingClientRect().width;
+        }
```

Comment:
I don't think this code handles hidden columns correctly. We don't create the `.header-container` element for hidden columns which means that your `headers` variable isn't guaranteed to be a parallel array to `this.table.columns`.

Reply:
Handling hidden columns now.